src/ray/raylet/task_dependency_manager.cc

```diff
@@ -169,7 +169,56 @@ bool TaskDependencyManager::SubscribeDependencies(
   return (task_entry.num_missing_dependencies == 0);
 }
 
-bool TaskDependencyManager::UnsubscribeDependencies(const TaskID &task_id) {
+void TaskDependencyManager::RemoveTaskDependency(const TaskID &task_id, const ObjectID &object_id) {
+  // Remove the task from the list of tasks that are dependent on this
+  // object.
+  // Get the ID of the task that creates the dependency.
+  TaskID creating_task_id = ComputeTaskId(object_id);
+  auto creating_task_entry = required_tasks_.find(creating_task_id);
+  std::vector<TaskID> &dependent_tasks = creating_task_entry->second[object_id];
+  auto it = std::find(dependent_tasks.begin(), dependent_tasks.end(), task_id);
+  RAY_CHECK(it != dependent_tasks.end());
+  dependent_tasks.erase(it);
+  // If the unsubscribed task was the only task dependent on the object, then
+  // erase the object entry.
+  if (dependent_tasks.empty()) {
+    creating_task_entry->second.erase(object_id);
+    // Remove the task that creates this object if there are no more object
+    // dependencies created by the task.
+    if (creating_task_entry->second.empty()) {
+      required_tasks_.erase(creating_task_entry);
+    }
+  }
+}
+
+bool TaskDependencyManager::UnsubscribeGetDependencies(const TaskID &task_id) {
+  // Remove the task from the table of subscribed tasks.
+  auto it = task_dependencies_.find(task_id);
+  if (it == task_dependencies_.end()) {
+    return false;
+  }
+
+  const TaskDependencies task_entry = std::move(it->second);
+
+  // Remove the task's dependencies.
+  for (const auto &object_id : task_entry.get_dependencies) {
+    RemoveTaskDependency(task_id, object_id);
```

Comment:
I think you also need to erase the `object_id`s from `task_entry.get_dependencies` (otherwise, it will appear as if the task still depends on these objects). You can do that with `task_entry.get_dependencies.clear();`, after this for loop.